optuna/integration/chainermn.py

```diff
@@ -169,8 +160,7 @@ class ChainerMNTrial(BaseTrial):
             index.html#communicators>`_.
     """
 
-    def __init__(self, trial, comm):
-        # type: (Optional[Trial], CommunicatorBase) -> None
+    def __init__(self, trial: "Optional[Trial]", comm: "CommunicatorBase") -> None:
```

Comment:
I think `Trial` will not cause loop reference.
```suggestion
    def __init__(self, trial: Optional[Trial], comm: "CommunicatorBase") -> None:
```